### PR TITLE
refactor: extract screen management from main into a module 

### PR DESCRIPTION
**Pull request by fabiomaciel**
separate into a module, all the logics to create and update screen

### CODE REVIEW

File: src/screen.c [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+#include "screen.h"
+
+#include <stdint.h>
+#include <SDL.h>
+
+const int SCREEN_WIDTH  = 640;
+const int SCREEN_HEIGHT = 480;
+
+bool init_screen(screen_t* screen){
```
Comment (johnnyrichard) [inline src/screen.c]: We should follow a styling consistency among our structs.

a) We are using the struct name as function namespace.

      bool screen_init(screen_t* screen){
           ^^^^^^^^^^^

b) Every function definition should have the `{` starting from a new line. 

    bool screen_init(screen_t* screen)
    {
    ...
    }

File: src/screen.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+  if (SDL_RenderSetLogicalSize(screen->renderer, screen->width, screen->height) < 0) {
+    fprintf(stderr, "Could not set logical size: %s\n", SDL_GetError());
+    return false;
+  }
+
+  return true;
+}
+
+void reset_screen(screen_t *screen){
```
Comment (johnnyrichard) [inline src/screen.c]:
```suggestion
void screen_reset(screen_t *screen)
{
```

File: src/screen.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  SDL_SetRenderDrawColor(screen->renderer, 0, 0, 0, 1);
+  SDL_RenderClear(screen->renderer);
+}
+
+void terminate_screen(screen_t *screen){
```
Comment (johnnyrichard) [inline src/screen.c]: What do you think about call this function `destroy`?
```suggestion
void screen_destroy(screen_t *screen)
{
```

File: src/screen.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+  SDL_DestroyRenderer(screen->renderer);
+  SDL_DestroyWindow(screen->window);
+  SDL_Quit();
```
Comment (johnnyrichard) [inline src/screen.c]: I'm not sure if **SDL termination** (`SDL_Quit`) should be part of screen life cycle, what about move it up? any thought?

**Review by johnnyrichard — COMMENTED** [review]
Nice, just few adjustments.

Could you please express better the benefit of extracting the screen/window logic to a new file?
I think we can add more context to the commit message.

Another suggestion would be re-title the commit to:
    
    screen: Extract screen management object from main function


```
Reviewed-by: Johnny Richard <johnny@johnnyrichard.com>
```

File: src/screen.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+#include "screen.h"
+
+#include <stdint.h>
+#include <SDL.h>
+
+const int SCREEN_WIDTH  = 640;
+const int SCREEN_HEIGHT = 480;
+
+bool init_screen(screen_t* screen){
+
```
Comment (johnnyrichard) [inline src/screen.c:44]: How about add an assertion before access properties?
```suggestion
  assert(screen && "screen is required");
```

File: src/screen.c (hunk 1 above)
Comment (fabiomaciel) [reply src/screen.c]: not only brackets but type was in another line either, so i fixed to follow a standard.

File: src/screen.c (hunk 3 above)
Comment (fabiomaciel) [reply src/screen.c]: sure, this is something i didnt pay attention, but makes sense and now is fixed.

File: src/screen.c (hunk 4 above)
Comment (fabiomaciel) [reply src/screen.c]: yes, make no sense screen module works with non screen actions of sdl.

File: src/screen.c (hunk 5 above)
Comment (fabiomaciel) [reply src/screen.c:44]: done.

**Review by johnnyrichard — APPROVED** [review]
Great. Thanks!

**Pull request merged**